src/ASM/ASMmxBase.C

```diff
@@ -229,6 +229,70 @@ ASMmxBase::SurfaceVec ASMmxBase::establishBases(Go::SplineSurface* surf,
                                                                   u0,vg,XYZ1,ndim,
                                                                   false,XYZ1));
     geoBasis = 3;
+  } else if (type == SUBGRID) {
+    // basis1 should be one degree higher than basis2 and C^p-1 continuous
+    int ndim = surf->dimension();
+    result[1].reset(new Go::SplineSurface(*surf));
+    Go::SplineSurface tmp(*surf);
+    for (size_t i = 0; i < 2; ++i) {
+      RealArray extraKnots;
+      RealArray::const_iterator uit = tmp.basis(i).begin();
+      double ucurr, uprev = *(uit++);
+      while (uit != tmp.basis(i).end())
+      {
+        ucurr = *(uit++);
+        if (ucurr > uprev)
+          for (int i = 0; i < 1; i++)
```

Comment:
This is the smallest for-loop I ever laid eyes on. Either take some "number-of-subdivisions-per-element" parameter in, or consider removing it

Reply:
copy-paste from uniformRefine - quick solutions for quick living.. i can do better.